### ISSUE TITLE
GTK should be marked as initialized after `parent_startup()`

### ISSUE THREAD

**Issue by YaLTeR** [gtk-rs/gtk4-rs#236]
Right now it's marked as initialized only after the entire `startup()` in a `GtkApplication` subclass finishes, however you might reasonably want to do some GTK stuff after calling `parent_startup()`:

```rust
    fn startup(&self, self_: &Self::Type) {
            self.parent_startup(self_);

            let provider = gtk::CssProvider::new();
            provider.load_from_resource("/org/gnome/gitlab/YaLTeR/VideoTrimmer/style.css");
            gtk::StyleContext::add_provider_for_display(
                &gdk::Display::get_default().unwrap(),
                &provider,
                gtk::STYLE_PROVIDER_PRIORITY_APPLICATION,
            );
    }
```

GTK is initialized in `parent_startup()`: https://gitlab.gnome.org/GNOME/gtk/-/blob/cc64dbafbc8158d5116bf2afe2d422da48b77137/gtk/gtkapplication.c#L262

cc @bilelmoussaoui

**Comment by sdroege** [comment]
The same change will also be needed for gtk3

**Comment by YaLTeR** [comment]
To reproduce, add this to the [application subclass example](https://github.com/gtk-rs/gtk4-rs/blob/cb5896a15fe712ed3568b9a3e20d1b02d150faa7/examples/src/bin/application_subclass.rs#L25).

```rust
        fn startup(&self, application: &Self::Type) {
            self.parent_startup(application);

            let _ = gtk::Button::new();
        }
```

**Comment by sdroege** [comment]
The problem here is

```rust
unsafe extern "C" fn application_startup<T: ObjectSubclass>(ptr: *mut gio::ffi::GApplication)
where
    T: GtkApplicationImpl,
{
    let instance = &*(ptr as *mut T::Instance);
    let imp = instance.get_impl();
    let wrap: Borrowed<gio::Application> = from_glib_borrow(ptr);
    imp.startup(wrap.unsafe_cast_ref());
    crate::rt::set_initialized();
}
```

It would only be initialized after the `startup` implementation of the subclass. Somehow we need to hook into the `startup` of `GtkApplication` directly, the one that is called as part of `parent_startup` above.

**Comment by sdroege** [comment]
See https://github.com/gtk-rs/gtk4-rs/pull/833

**Comment by sdroege** [comment]
This is fixed in 0.4.3 now
